projects/Irdest/services/ratmand/module.nix

```diff
@@ -0,0 +1,107 @@
+{
+  config,
+  pkgs,
+  lib,
+  ...
+}:
+let
+  cfg = config.services.ratmand;
+
+  # Convert `cfg.settings` to arguments for `ratmand generate` to generate the config
+  # file.
+  configPatchArgs =
+    lib.concatLists (
+      (lib.mapAttrsToList (
+        section: sectionAttrs:
+        lib.mapAttrsToList (
+          key: value:
+          ''--patch "${section}/${key}=${
+            if lib.isBool value then lib.boolToString value else toString value
+          }"''
+        ) sectionAttrs
+      ) (cfg.settings // { ratmand = lib.removeAttrs (cfg.settings.ratmand or { }) [ "peers" ]; }))
+    )
+    ++ lib.map (peer: ''--add-peer "${peer}"'') cfg.settings.ratmand.peers;
+
+  configFile = pkgs.runCommand "ratmand-config" { } ''
+    ${lib.getExe' cfg.package "ratmand"} --config "$out" generate ${lib.concatStringsSep " " configPatchArgs}
+  '';
```

Comment:
This might eventually be replaced by a kdl format once that lands in `pkgs.formats` right? Could you link to the corresponding Nixpkgs issue? Without having looked at the specific differences between kdl and nix, this sounds like a more stable option than to use their CLI to generate config.